src/rest/connector/acisdk_implementation.py

```diff
@@ -0,0 +1,285 @@
+        session = LoginSession(self.url, username, password, timeout=timeout)
+
+        MoDirectory = getattr(import_module('cobra.mit.access'), 'MoDirectory')
+        self.mo_dir = MoDirectory(session)
+
+        log.info("Connecting to '{d}' with alias "
+                 "'{a}'".format(d=self.device.name, a=self.alias))
+
+        self.mo_dir.login()
+
+        self._is_connected = True
+        log.info("Connected successfully to '{d}'".format(d=self.device.name))
+
+    def disconnect(self):
+        """disconnect the device for this particular alias"""
+
+        log.info("Disconnecting from '{d}' with "
+                 "alias '{a}'".format(d=self.device.name, a=self.alias))
+        try:
+            self.mo_dir.logout()
+        finally:
+            self._is_connected = False
+        log.info("Disconnected successfully from "
+                 "'{d}'".format(d=self.device.name))
+
+    def is_connected(func):
+        """Decorator to make sure session to device is active
+           There is limitation on the amount of time the session cab be active
+           on the APIC. However, there are no way to verify if
+           session is still active unless sending a command. So, its just
+           faster to reconnect every time.
+        """
+        @wraps(func)
+        def decorated(self, *args, **kwargs):
+            # Check if connected
+            try:
+                ret = func(self, *args, **kwargs)
+            except Exception as e:
+                if not hasattr(e, 'reason') or e.reason != 'Token was invalid (Error: Token timeout)':
+                    raise
+
+                self.disconnect()
+
+                if 'timeout' in kwargs:
+                    self.connect(timeout=kwargs['timeout'])
+                else:
+                    self.connect()
+
+                ret = func(self, *args, **kwargs)
+            return ret
+
+        return decorated
+
+    def log_action(func):
+        """Decorator to log the actions made by MoDirectory functions.
+        """
+        @wraps(func)
+        def decorated(self, *args, **kwargs):
+
+            ret = func(self, *args, **kwargs)
+            log.info(f'Output of function {func.__name__} is:\n{pformat(ret, indent=2)}')
+            return ret
+
+        return decorated
+
+    @BaseConnection.locked
+    @is_connected
+    @log_action
+    def query(self, queryObject):
+        """
+        Queries the MIT for a specified object. The queryObject provides a
+        variety of search options.
+        """
+        return self.mo_dir.query(queryObject)
+
+    @BaseConnection.locked
+    @is_connected
+    @log_action
+    def commit(self, configObject, sync_wait_timeout=None):
+        """
+        Short-form commit operation for a configRequest
+        """
+        return self.mo_dir.commit(configObject, sync_wait_timeout=sync_wait_timeout)
+
+    @BaseConnection.locked
+    @is_connected
+    @log_action
+    def lookupByDn(self, dnStrOrDn, **queryParams):
```

Comment:
we don't use this style func name. please change to `lookup_by_dn`. and please do the same where applicable.

same for argument. `dnStrOrDn` -> `dn_str_or_dn`

Reply:
Function and argument names are mimicked from the APIC SDK (Cobra) connector functions, and are instantiated here for two reasons:
1. use of log_action decorator (logs the result of the function)
2. ease of use (`output = device.connection.function` would call the SDK (Cobra) corresponding function while preserving the names and variables)
In this way, someone accustomed to the SDK would be able to use them easier.
Can still change them if needed, of course.

Reply:
if reasonable reason, good. thanks for the explanation. good idea/way. as reference, in doc or as comment in code, better to mention it's same with SDK(Cobra) and for ease of use.